### PR TITLE
Migrating to axum Web server.

### PR DESCRIPTION
**Pull request by kevinswiber**
Added support for previous functionality.  Changed status code implementation to exactly match the original httpbin project.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    routing::get,
+    Router,
+};
+use minijinja::render;
+use std::env;
+use std::net::SocketAddr;
+
+const INDEX_TEMPLATE: &'static str = include_str!("./templates/index.html");
+const OPENAPI_SPECIFICATION: &'static str = include_str!("./templates/openapi.yaml");
+const API_DOCS_LOCATION: &'static str = "https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/postman-open-technologies/httpbin-rs/main/src/templates/openapi.yaml&poo=1&nocors";
```
Comment (MikeRalphson) [inline src/main.rs]: Can we drop `poo=1&`  here? I was trying to bust a  cache.

File: src/routes/status_codes.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                | StatusCode::TEMPORARY_REDIRECT => builder
+                    .header(header::LOCATION, REDIRECT_LOCATION)
+                    .body(body)
+                    .unwrap(),
+                StatusCode::UNAUTHORIZED => builder
+                    .header(header::WWW_AUTHENTICATE, "Basic realm=\"Fake realm\"")
+                    .body(body)
+                    .unwrap(),
+                StatusCode::PAYMENT_REQUIRED => {
+                    body = "Fuck you, pay me!".to_string(); // 👀 https://github.com/postmanlabs/httpbin/blob/master/httpbin/helpers.py#L221
```
Comment (MikeRalphson) [inline src/routes/status_codes.rs]: We can drop compatibility here I think, we can use a quote... Show me the money etc.

**Review by MikeRalphson — CHANGES_REQUESTED** [review]
Thanks, this looks amazing. Couple of nits. Do you have a preferred framework to port the tests to?

**Comment by kevinswiber** [comment]
Resolved the nits and added tests, which inspired a little refactoring.  @MikeRalphson Please have another look when you get a chance!

**Review by MikeRalphson — APPROVED** [review]
LGTM, thank you so much.

File: src/routes/status_codes.rs (hunk 2 above)
Comment (doc-jones) [reply src/routes/status_codes.rs]: salty for a work project

**Review by doc-jones — APPROVED** [review]

File: src/routes/status_codes.rs (hunk 2 above)
Comment (kevinswiber) [reply src/routes/status_codes.rs]: @doc-jones Haha. For sure. I was just copying exactly what’s in the original Python project… which is also technically a work project. I changed it to Jerry Maguire per the suggestion.

**Pull request merged**